packages/main/src/plugin/kubernetes/contexts-manager-experimental.ts

```diff
@@ -93,6 +148,15 @@ export class ContextsManagerExperimental {
     return result;
   }
 
+  /* getPermissions returns the current permissions */
+  getPermissions(): Map</* contextName */ string, Map</* resource */ string, ContextResourcePermission>> {
```

Comment:
Please create a interface for the result something with worded accessors, `Map<string, Map<string, ContextResourcePermission>>` is not friendly;

Ideally when dealing with kubernetes cluster I want objects, for example if I can list pods I want to do something like

```ts
// checking if I can access pods in a given namespace
KubernetesClient#getClusterConnection(context)#getPermission({ resource: "pods", namespace: "default" }): ContextResourcePermission
// checking all the permissions
KubernetesClient#getClusterConnection(context)#listPermissions(): ContextResourcePermission[]
```

Reply:
We don't need to Map resource to V1ResourceAttributes since it contains the resource name

Reply:
I would prefer not focus too much on this part for now, it will most probably change when we will dispatch this data to the frontend, in a follow-up PR